packages/gatsby-link/src/index.js

```diff
@@ -17,6 +17,12 @@ function normalizePath(path) {
   return path.replace(/^\/\//g, `/`)
 }
 
+function createLocation(path, history) {
+  var location = cL(path, null, null, history.location)
```

Comment:
I think the problem was i just assuming we were calling these two functions with the same arguments. but it looks like the RR createLocation has a different signature.

Reply:
yeah, it seems that this local `createLocation` signature is different than one imported from `history`, I will update this PR and instead of reverting your changes will adjust params in `createLocation` calls